src/errors/validation_exception.rs

```diff
@@ -45,12 +45,16 @@ impl ValidationError {
         }
     }
 
-    pub fn display(&self, py: Python) -> String {
-        let count = self.line_errors.len();
-        let plural = if count == 1 { "" } else { "s" };
-        let title: &str = self.title.extract(py).unwrap();
+    pub fn display(&self, py: Python, prefix_override: Option<&'static str>) -> String {
         let line_errors = pretty_py_line_errors(py, self.line_errors.iter());
-        format!("{count} validation error{plural} for {title}\n{line_errors}")
+        if let Some(prefix) = prefix_override {
+            format!("{prefix}\n{line_errors}")
+        } else {
+            let count = self.line_errors.len();
+            let plural = if count == 1 { "" } else { "s" };
+            let title: &str = self.title.extract(py).unwrap();
+            format!("{count} validation error{plural} for {title}\n{line_errors}")
+        }
```

Comment:
If there's a cleaner way to do this (presumably that doesn't involve creating intermediate `String`s), I'm happy to change it (and learn something).

Reply:
Yeah I knew this didn't but I was trying to find a way to not duplicate the inclusion of `line_errors` without inserting string creation. Like, if this was python code I guess I'd find it a big ugly (definitely splitting hairs at this point though); wasn't sure if there might be a better pattern for doing this in Rust. Anyway if it looks good to you then I'm happy.